src/vellum/workflows/expressions/greater_than.py

```diff
@@ -38,4 +38,9 @@ def resolve(self, state: "BaseState") -> bool:
         if not has_gt(lhs):
             raise InvalidExpressionException(f"'{lhs.__class__.__name__}' must support the '>' operator")
 
-        return lhs > rhs
+        try:
+            return lhs > rhs
+        except TypeError as e:
+            raise InvalidExpressionException(
+                f"Cannot compare '{type(lhs).__name__}' with '{type(rhs).__name__}': {str(e)}"
```

Comment:
We can reduce the amount of diffs in this PR and redundancy in the error message by just using `str(e)` as the full message